src/Core/System/Snippet/Command/Util/CountryAgnosticFileValidator.php

```diff
@@ -0,0 +1,175 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Snippet\Command\Util;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\Snippet\SnippetException;
+use Shopware\Core\System\Snippet\SnippetPatterns;
+use Shopware\Core\System\Snippet\Struct\TranslationFile;
+use Shopware\Core\System\Snippet\Struct\TranslationFileCollection;
+use Shopware\Core\System\Snippet\Struct\ValidatedTranslationFileOptions;
+use Shopware\Core\System\Snippet\Struct\ValidatedTranslationFileStruct;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @internal
+ */
+#[Package('discovery')]
+class CountryAgnosticFileValidator
+{
+    public const CORE_DOMAINS = [
+        'administration' => 'Administration',
+        'messages' => 'Core',
+        'storefront' => 'Storefront',
+    ];
+
+    public function getFinder(ValidatedTranslationFileOptions $options): Finder
+    {
+        $finder = (new Finder())
+            ->files()
+            ->ignoreUnreadableDirs()
+            ->ignoreDotFiles(true)
+            ->ignoreVCS(true)
+            ->exclude([
+                'node_modules',
+                'vendor',
+                'bin',
+                'static',
+                // Translations of languages fetched from crowdin should not be validated
+                'SwagLanguagePack/src/Resources/snippet',
+                'SwagLanguagePack/src/Resources/app/administration/src/snippet',
+                ...$options->getIgnoredPaths(),
+            ])
+            ->name([SnippetPatterns::CORE_SNIPPET_FILE_PATTERN, SnippetPatterns::ADMIN_SNIPPET_FILE_PATTERN])
+            ->sortByName(true);
+
+        if ($options->getDir()) {
+            $finder->in($options->getDir());
+        } elseif (empty($options->getExtensionPaths())) {
+            $finder->in('src');
+
+            if ($options->isAll()) {
+                $finder->in('custom/plugins');
+                $finder->in('custom/apps');
+            }
+        } else {
+            $extensionPaths = \array_reduce($options->getExtensionPaths(), static function (array $accumulator, string $extension): array {
```

Comment:
to improve readability of this method, you could move this part into an own private method:

```
$extensionPaths = $this->resolveExtensionPaths(...);
$finder->in($extensionPaths);
```

Reply:
this is in general very hard to read code. I need a few minutes to know, what you are trying to achieve. and I am not sure, if I understood correctly. 

you want to have the paths of extensions, right? I wonder why you do not use the "path" value which apps and plugins have in their entity :thinking: